### PR TITLE
Io issues

### PR DESCRIPTION
**Pull request by Silv3rcircl3**

### CODE REVIEW

File: src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs [hunk 1]
```diff
@@ -231,7 +283,14 @@ private void SendData(ByteString data)
         {
             Send(() =>
             {
-                _dataQueue.Add(data);
+                try
+                {
+                    _dataQueue.Add(data);
+                }
+                catch (Exception ex)
```
Comment (Horusiath) [inline src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs]: > Yo dawg, I've heard that you like IOExceptions, so we'll catch your IOException and wrap it with IOException.

I belive, this should be `catch(Exception ex) when !(ex is IOException)` ;)

File: src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs [hunk 2]
```diff
@@ -245,24 +304,35 @@ private void SendMessage(IAdapterToStageMessage msg, bool handleCancelled = true
...
-                    throw PublisherClosedException;
+                    _stageWithCallback.WakeUp(msg).Wait(_writeTimeout);
+                    if (_downstreamStatus.Value is Canceled && handleCancelled)
+                    {
+                        //Publisher considered to be terminated at earliest convenience to minimize messages sending back and forth
+                        _isPublisherAlive = false;
+                        throw PublisherClosedException;
+                    }
+                }
+                catch (IOException)
```
Comment (Horusiath) [inline src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs]: Same as above.

File: src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs (hunk 1 above)
Comment (Silv3rcircl3) [inline src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs]: I wanted to stay as close as possible to the jvm  ;-) 
https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamSourceStage.scala#L147

Should I report this back?

Comment (Horusiath) [inline src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs]: I think, the canonical impl just didn't care about that. Personally I would vote for avoiding IOException inception.

**Comment by Silv3rcircl3** [comment]
@Horusiath addressed your remarks

**Pull request merged**